Scripts/validate-partner-version.rb

```diff
@@ -0,0 +1,12 @@
+# Validates a partner version string checking it is well-formed.
+
+require_relative 'common'
+
+PARTNER_VERSION_REGEX = /^(?>~> )?[0-9]+(?>\.[0-9]+){1,}$/
+
+# Parse the version string from the arguments
+abort "Missing argument. Requires: version string." unless ARGV.count == 1
```

Comment:
I see there's a mixture of `abort` and `exit` in these scripts. Might actually just be easier to use `abort`, since we should get a more useful message in the log, but not a huge deal right now.